--- a/Arm/Insts/DPI/Logical_imm.lean
+++ b/Arm/Insts/DPI/Logical_imm.lean
@@ -117,17 +117,7 @@
     | none => write_err (StateError.Illegal s!"Illegal {inst} encountered!") s
     | some (imm, _) =>
       let op := decode_op inst.opc
-      let operand1 :=
-        match op with
-        | .ORR =>
-          -- (TODO): a `dsimproc` for MoveWidePreferred?
-          if  ¬ MoveWidePreferred inst.sf inst.N inst.imms inst.immr then
-             -- MOV (bitmask immediate) is an alias of ORR when
-             -- Rd == '11111' and Move Wide is not preferred.
-             read_gpr_zr datasize inst.Rn s
-          else
-            read_gpr datasize inst.Rn s
-        | _ => read_gpr datasize inst.Rn s
+      let operand1 := read_gpr_zr datasize inst.Rn s
       let (result, maybe_pstate) := exec_logical_imm_op op operand1 imm
       -- State Updates
       let s'            := write_pc ((read_pc s) + 4#64) s
